content/800-data-platform/200-pulse/400-api-reference.mdx

```diff
@@ -0,0 +1,287 @@
+---
+title: 'API Reference'
+metaTitle: 'API Reference'
+metaDescription: 'API reference documentation for Pulse.'
+tocDepth: 4
+toc: true
+---
+
+<TopBlock>
+
+The Pulse API reference documentation is based on the following schema:
+
+```prisma
+model User {
+  id    Int     @id @default(autoincrement())
+  name  String?
+  email String  @unique
+}
+```
+
+All example are based on the `User` model.
+
+</TopBlock>
+
+## <inlinecode>subscribe</inlinecode>
+
+`subscribe` returns all data change events related to the table you call this method on:
+
+```ts
+const subscription = await prisma.user.subscribe()
+```
+
+### Remarks
+
+Passing an empty object to the `subscribe` function will also behaves the same as passing no argument to the function:
+
+```ts
+const subscription = await prisma.user.subscribe({})
+```
+
+### Database events filters
+
+Inside of the `subscribe()` method you can use filters to get specific events and data. The three filter options currently supported are described below:
+
+#### Create
+
+You can use the `create` filter to retrieve all _create_ events on a table. A create event is triggered when a new record is created in a table.
+
+Pulse returns the values of the new record as an object named `after`.
+
+##### Example
+
+<CodeWithResult outputResultText="event">
+<cmd>
+
+```ts
+const subscription = await prisma.user.subscribe({
+  create: {},
+})
+```
+
+</cmd>
+<cmdResult>
+
+```ts no-copy
+{
+  action: 'create',
+  after: {
+    id: 1,
+    email: 'marc@prisma.io',
+    name: 'Marc'
+  }
+}
+```
+
+</cmdResult>
+</CodeWithResult>
+
+#### Update
+
+You can use the `update` filter to retrieve all _update_ events on a table. An update event is triggered when a value in a database record has changed.
+
+Pulse returns the values of the changed record as an object named `after`. This represents the state of the record _after_ the change event.
```

Comment:
does update not also contain `before`?

Reply:
It doesn't at the moment.

Reply:
Correct, it won't return before by default.